File: frontend/src/types/patient.ts
/**
 * Patient entity representing a patient.
 */
export interface Patient {
  id: string;
  facilityId: string;
  keycloakUserId?: string;
  name: string;
  birthDate: string;
  email?: string;
  phone?: string;
  address?: string;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}

/**
 * Request DTO for creating a patient.
 */
export interface CreatePatientRequest {
  facilityId: string;
  keycloakUserId?: string;
  name: string;
  birthDate: string;
  email?: string;
  phone?: string;
  address?: string;
  active?: boolean;
}

/**
 * Request DTO for updating a patient.
 */
export interface UpdatePatientRequest {
  name?: string;
  birthDate?: string;
  email?: string;
  phone?: string;
  address?: string;
  active?: boolean;
}

File: backend/src/main/java/com/smiles/patients/domain/Patient.java
package com.smiles.patients.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

/**
 * Patient entity representing a patient registered at a facility.
 */
@Entity
@Table(name = "patient")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "facility_id", nullable = false)
    private UUID facilityId;

    @Column(name = "keycloak_user_id", unique = true)
    private String keycloakUserId;

    @Column(nullable = false)
    private String name;

    @Column(name = "birth_date", nullable = false)
    private LocalDate birthDate;

    @Column
    private String email;

    @Column(length = 50)
    private String phone;

    @Column(length = 500)
    private String address;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}
